user/permissions.py

```diff
@@ -0,0 +1,13 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+from rest_framework.exceptions import NotAuthenticated
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    def has_permission(self, request, view):
+        if not request.user.is_authenticated:
+            raise NotAuthenticated()

```

Comment:
Raising a `NotAuthenticated` exception directly in the `has_permission` method might not be ideal, as it could lead to inconsistent behavior across different views. Consider returning `False` instead, which allows the framework to handle authentication failures more uniformly.